lib/Hakyll/Core/Runtime.hs

```diff
@@ -321,13 +319,21 @@ chase id' = do
                 let depDone =
                         depId `S.member` done ||
                         (depId, depSnapshot) `S.member` snapshots
+                    actualDep = [(depId, depSnapshot) | not depDone]
 
-                -- return values to add to runtimeDependencies
-                pure $ if depDone then [] else [depId]
+                return actualDep
 
             modifyRuntimeState $ \s -> s
                 { runtimeTodo         = M.insert id'
-                    (if null deps' then c else compilerResult result)
+                    (if null deps then c else compilerResult result)
                     (runtimeTodo s)
-                , runtimeDependencies = M.insertWith S.union id' (S.fromList deps') deps
+                 -- We track dependencies only to inform users when an infinite loop is detected
+                , runtimeDependencies = M.insertWith S.union id' (S.fromList deps) (runtimeDependencies s)
```

Comment:
I think this should *replace* the value in the map, not *extend* it with new data. Also, we should completely clear the map at each iteration of `pickAndChase`.

The reason is: by the time we stumble into a dependency cycle, we might have already accumulated a ton of entries in this map. But the cycle can only be formed by something that is *still* running, so we only care about the dependencies of the compilers that ran during the current invocation of `chase`.

Does that make sense, or am I missing something?

Reply:
Hmm I see what you mean. My issue is: what happens when two threads replace the map at the same time?

It might be preferable to remove the entries in the `runtimeDependencies` map when compiling an item results in `CompileDone`?

Reply:
I don't think they can: each thread is working on its own identifier, so they will never try to replace the same key in the map.

Also, we can put the map into the `Idled` constructor. That'll take care of "clearing the map" (we'll get a new one every time), avoid shared state, and make it clear where and why we need the map at all.

Reply:
Oh, you already implemented deletion. It's up to you then; I'm okay merging this unless you like my idea so much that you're willing to try it out :)

Reply:
Yeah actually I like removing entirely the `runtimeDependencies` field. I'll try it out your way tonight

Reply:
In the end I like the fact that `Progress` is a monoid. I'll leave it as it for now